src/Microsoft.VisualStudio.Editors/PropPages/PackagePropPage.vb

```diff
@@ -230,7 +285,113 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
         ''' </summary>
         Private Sub NeutralLanguageComboBox_DropDown(sender As Object, e As EventArgs) Handles NeutralLanguageComboBox.DropDown
             PopulateNeutralLanguageComboBox(NeutralLanguageComboBox)
-            Common.SetComboBoxDropdownWidth(NeutralLanguageComboBox)
+            SetComboBoxDropdownWidth(NeutralLanguageComboBox)
+        End Sub
+
+        ''' <summary>
+        '''  Set the radio button selection and update both textboxes
+        ''' </summary>
+        ''' <param name="setLicenseExpression">Sets the radio button for LicensesExpression, if false sets to LicenseFile</param>
+        Private Sub SetLicenseRadioButtons(setLicenseExpression As Boolean)
+            If (Not _newLicensePropertyDetectedAtInit) Then
+                LicenseTypeFirstSelected()
+                _newLicensePropertyDetectedAtInit = False
+            End If
+            LicenseFileRadioButton.Checked = Not setLicenseExpression
+            LicenseExpressionRadioButton.Checked = setLicenseExpression
+            PackageLicenseExpression.Enabled = setLicenseExpression
+            LicenseFileNameTextBox.Enabled = Not setLicenseExpression
+            LicenseBrowseButton.Enabled = Not setLicenseExpression
+            If setLicenseExpression AndAlso Not LicenseFileNameTextBox.Text = "" Then
+                SetCommonPropertyValue(GetPropertyDescriptor("PackageLicenseFile"), "")
+                LicenseFileNameTextBox.Text = ""
+            ElseIf (Not PackageLicenseExpression.Text = "") Then
```

Comment:
I could be reading this wrong but if setLicenseExpression is true and the LicenseFileNameTextBox is empty, won't this clear an existing value in the license expression text box? Seems like thats unwanted if setLicenseExpression is true.

Reply:
You are correct. I wasn't seeing that in my tests because there would never be any text in PackageLicenseExpression if setLicenseExpression was true because it was already handled. But, I am going to change the check so it can be used in the future without confusion.